script.js

```diff
@@ -1,4 +1,118 @@
+var playerList = [];
+var gameStatus = 'register';
+// var numOfDice = 0;
+// var numOfGame = 0;
+var gameMode = 'default'
+var scoreR1 =[];
+var playerCounter = 0;
+var winnerCounter = 1;
+var winnerNum = 0;
+var myOutputValue = '';
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (gameStatus == 'register'){
+    if (!input){
+      myOutputValue = `Welcome to 🎲 Beat That 🎲 <br><br>
+                      Please submit player ${playerList.length + 1} name.`
+    }
+    if (input && input.toLowerCase() != 'start'){
+      playerList.push(input);
+      console.log (`'registered players :${playerList}`); 
+      myOutputValue = `Welcome Player ${playerList.length}: ${input}! <br><br>
+                      Please submit player ${playerList.length + 1} name <br><br>
+                      OR <br><br>
+                      Enter "start" to kick start the game!`
+    }  
+    if (input.toLowerCase() == 'start'){
+      gameStatus = 'select mode';
+      myOutputValue = `The registered player(s) in the game are: <br>
+                      ${playerList} <br><br>
+                      Please select the game mode: <br>
+                      [ 1 ] Bigger number wins<br>
+                      [ 2 ] Smaller number wins`;
+    }
+  }
+
+    if (gameStatus == 'select mode'){
+      if (input == 1){
+      gameMode = 'big';
+      gameStatus = 'start';
+    }
+    if (input == 2){
+      gameMode = 'small';
+      gameStatus = 'start';
+    }
+    if (!input){
+      myOutputValue =`Please select the game mode: <br>
+                      [ 1 ] Bigger number wins<br>
+                      [ 2 ] Smaller number wins`;
+    }
+  }
```

Comment:
You can consider using helper functions once you change game modes, instead of putting everything into the main function